src/opening-period/rule/Rule.tsx

```diff
@@ -0,0 +1,303 @@
+  control: Control;
+  setValue: Function;
+  remove: Function;
+  register: Function;
+  index: number;
+  ruleConfig: UiRuleConfig;
+}): JSX.Element {
+  const {
+    id,
+    context,
+    subject,
+    start: startAt,
+    frequency_modifier: frequencyModifier = null,
+    frequency_ordinal: frequencyOrdinal = null,
+  } = rule;
+
+  const {
+    context: { options: contextOptions },
+    subject: { options: subjectOptions },
+    frequencyModifier: { options: frequencyModifierOptions },
+  } = ruleConfig;
+
+  const selectedSubject = subjectOptions.find(
+    ({ value }: InputOption) => value === `${subject}`
+  );
+  const [subjectLabel, setSubjectLabel] = useState<string>(
+    selectedSubject?.label ?? ''
+  );
+
+  const currentFrequency = {
+    frequency_modifier: frequencyModifier,
+    frequency_ordinal: frequencyOrdinal,
+  } as Frequency;
+
+  const knownFrequencyValues: FrequencyOption[] = [
+    ...hardCodedFrequencyOptions,
+    ...frequencyModifierOptions.map(
+      (modifierOption: InputOption): FrequencyOption => ({
+        label: modifierOption.label,
+        value: {
+          frequency_ordinal: null,
+          frequency_modifier: modifierOption.value,
+        },
+      })
+    ),
+  ];
+
+  const isKnownFrequencySelected = knownFrequencyValues.find(
+    ({ value }) =>
+      value.frequency_modifier === currentFrequency.frequency_modifier &&
+      value.frequency_ordinal === currentFrequency.frequency_ordinal
+  );
+
+  const allFrequencyValues: FrequencyOption[] = isKnownFrequencySelected
+    ? knownFrequencyValues
+    : [
+        ...knownFrequencyValues,
+        {
+          label: generateUnknownFrequencyLabel(
+            currentFrequency.frequency_ordinal,
+            frequencyModifierOptions.find(
+              (modifierOption: InputOption) =>
+                modifierOption.value === currentFrequency.frequency_modifier
+            )?.label
+          ),
+          value: currentFrequency,
+        },
+      ];
+
+  const frequencyOptions: InputOption[] = allFrequencyValues.map(
+    convertFrequencyToInputOption
+  );
+
+  const frequencyModifierField = `rules[${index}].frequency_modifier`;
+  const frequencyOrdinalField = `rules[${index}].frequency_ordinal`;
+
+  const onFrequencyChange = (selected: InputOption): void => {
+    const selectedFrequency = allFrequencyValues.find(
+      ({ label }) => label === selected.label
+    );
+
+    setValue(
+      frequencyModifierField,
+      selectedFrequency?.value.frequency_modifier || null
+    );
+    setValue(
+      frequencyOrdinalField,
+      selectedFrequency?.value.frequency_ordinal || null
+    );
+  };
+
+  useEffect(() => {
+    register({ name: frequencyModifierField });
+    register({ name: frequencyOrdinalField });
+
+    // "If you choose to not use Controller and manually register fields, you will need to update the input value with setValue." - https://react-hook-form.com/api#register
```

Comment:
True, but I think this only applies in situations where Ref is not available, if I understood the documentation correct?

Reply:
Yes, and we cannot use ref since frequency is derived of two api values: modifier and ordinal

Reply:
It seems that react-hook-form does not initialize the value without using ref. At least I did not find that from the source code: https://github.com/react-hook-form/react-hook-form/blob/master/src/useForm.ts#L908